### PR TITLE
Remove instructions around milestones

### PR DESCRIPTION
**Pull request by albertteoh**
Signed-off-by: albertteoh <see.kwang.teoh@gmail.com>

## Short description of the changes
- Proposal to remove instructions around milestones. The current milestones are 4 years old and I don't think we have any plans on defining a milestone per release.

### CODE REVIEW

**Review by pavolloffay — DISMISSED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -13,18 +13,17 @@
         ```
       * Even if a submodule does not have a new release, it should be checked to see if there were any changes warranting cutting a new release and then including it.
     * Rotate the below release managers table placing yourself at the bottom. The date should be the first Wednesday of the month.
-2. Add all merged pull requests to the milestone for the release and create a new milestone for a next release e.g. `Release 1.16`.
```
Comment (pavolloffay) [inline RELEASE.md:16]: This section should be removed as well https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING.md#merging-prs

Comment (albertteoh) [reply RELEASE.md:16]: Thanks @pavolloffay; just to check, do you mean we should delete the entire "Merging PRs" section or just the last two points:

```
* pull request is assigned to the current release milestone
* add `changelog:*` and other labels
```

Comment (yurishkuro) [reply RELEASE.md:16]: just deleting these two bullets is fine

**Comment by albertteoh** [comment]
Closed in favour of https://github.com/jaegertracing/jaeger/pull/3859.

**Pull request closed without merging**